### PR TITLE
ENCD-6161-fix genome browser pinned files in sessionStorage

### PR DESCRIPTION
**Pull request by pedrohr**

### CODE REVIEW

File: src/encoded/static/components/region_search.js [hunk 1]
```diff
@@ -324,10 +324,68 @@ const RegionSearch = (props, context) => {
...
+            const interval = coords.split(':')[1];
+
             const coordinatesObj = {
-                contig: coords.split(':')[0],
-                x0: coords.split(':')[1].split('-')[0],
-                x1: coords.split(':')[1].split('-')[1],
+                contig: `chr${contig.split('chr')[1].toUpperCase()}`,
+                x0: interval.split('-')[0],
+                x1: interval.split('-')[1],
+                pinnedFiles: pinnedFilesAssembly[selectedAssembly],
```
Comment (zoldello) [inline src/encoded/static/components/region_search.js:388]: `selectedAssembly` is not guaranteed to be a value in `pinnedFilesAssembly` based on line 321.

File: src/encoded/static/components/genome_browser.js [hunk 2]
```diff
@@ -108,9 +108,9 @@ const getDefaultCoordinates = (assembly, annotation, ignoreCache = false) => {
 
     if (gVState && gVAssembly === assembly && gVAnnotation === annotation) {
         const savedState = gVState ? JSON.parse(gVState) : {};
-        ({ contig, x0, x1 } = savedState);
-    }
-    if (assembly === 'GRCh38') {
+        ({ contig } = savedState);
```
Comment (zoldello) [inline src/encoded/static/components/genome_browser.js:111]: Is this a good idea - 

`{x0, x1, pinnedFiles, contig } = savedState);`

File: src/encoded/static/components/genome_browser.js [hunk 3]
```diff
@@ -108,9 +108,9 @@ const getDefaultCoordinates = (assembly, annotation, ignoreCache = false) => {
...
+        ({ x0, x1, pinnedFiles } = savedState);
```
Comment (zoldello) [inline src/encoded/static/components/genome_browser.js:112]: You are not using `pinnedFiles`. It gets overwritten every time.

**Review by zoldello — CHANGES_REQUESTED** [review]
Let's have a conversation after you look at my comments.

File: src/encoded/static/components/genome_browser.js (hunk 3 above)
Comment (zoldello) [reply src/encoded/static/components/genome_browser.js:112]: This is major. The other comments are minor/insignificant.

File: src/encoded/static/components/region_search.js (hunk 1 above)
Comment (pedrohr) [reply src/encoded/static/components/region_search.js:388]: the interface only shows `mm10` and `GRCh38`

File: src/encoded/static/components/genome_browser.js (hunk 2 above)
Comment (pedrohr) [reply src/encoded/static/components/genome_browser.js:111]: yes, this is def better. I'm just reverting the file back to what it was, we should add this next time we change this lib

File: src/encoded/static/components/genome_browser.js (hunk 3 above)
Comment (pedrohr) [reply src/encoded/static/components/genome_browser.js:112]: which line are they getting overwritten?

Comment (zoldello) [reply src/encoded/static/components/genome_browser.js:112]: Never mind, I get it now.

**Review by zoldello — DISMISSED** [review]
Looks fine. Let's test it.

File: src/encoded/region_search.py [hunk 4]
```diff
@@ -215,7 +215,7 @@ def region_search(context, request):
         return result
 
     expand = True
-    if region and (re.match(r'chr\w:\d+-\d+', region) or re.match(r'rs\d+', region)):
+    if region and (re.match(r'chr\w+:\d+-\d+', region) or re.match(r'rs\d+', region)):
```
Comment (zoldello) [inline src/encoded/region_search.py:218]: Makes sense.

**Review by zoldello — APPROVED** [review]
Looks good. The CI test should pass as nothing affecting them changed.

**Pull request merged**